bin/liquidator/src/unwrappable_oracle_error.rs

```diff
@@ -0,0 +1,126 @@
+use anchor_lang::error::Error::AnchorError;
+use mango_v4::error::MangoError;
+use mango_v4::state::TokenIndex;
+use regex::Regex;
+
+pub trait UnwrappableOracleError {
+    fn try_unwrap_oracle_error(&self) -> Option<(TokenIndex, String)>;
+}
+
+impl UnwrappableOracleError for anyhow::Error {
+    fn try_unwrap_oracle_error(&self) -> Option<(TokenIndex, String)> {
+        let root_cause = self
+            .root_cause()
+            .downcast_ref::<anchor_lang::error::Error>();
+
+        if root_cause.is_none() {
+            return None;
+        }
+
+        if let AnchorError(ae) = root_cause.unwrap() {
+            let is_oracle_error = ae.error_code_number == MangoError::OracleConfidence.error_code()
+                || ae.error_code_number == MangoError::OracleStale.error_code();
+
+            if !is_oracle_error {
+                return None;
+            }
+
+            let error_str = ae.to_string();
+            return parse_oracle_error_string(&error_str);
+        }
+
+        return None;
```

Comment:
very minor, but the value of the last statement in a function is returned automatically if there's no semicolon, so ideomatically you'd just write `None` here